src/faces.jl

```diff
@@ -519,6 +523,11 @@ Base.merge(a::Face, b::Face, others::Face...) = merge(merge(a, b), others...)
 
 ## Getting the combined face from a set of properties ##
 
+# Putting these inside `getface` causes the julia compiler to box it
```

Comment:
I don't actually know why this happens, it's just that doing this change fixes the issue reported by `@code_warntype`... I'm kind of scared of closures for reasons like this :(